Assets/Scripts/Models/Area/RoomBehavior.cs

```diff
@@ -387,16 +387,15 @@ public RoomBehavior Clone()
         public void Control(Room room)
         {
             this.Room = room;
-            List<Tile> innerTiles = room.GetInnerTiles();
-            List<Tile> borderTiles = room.GetBorderingTiles();
-
-            List<Tile> allTiles = innerTiles.Union(borderTiles).ToList();
+            HashSet<Tile> innerTiles = room.GetInnerTiles();
+            HashSet<Tile> borderTiles = room.GetBoundaryTiles();
+            innerTiles.UnionWith(borderTiles);
```

Comment:
If innerTiles and borderTiles are only used to create a list of all tiles, is there a way to do it without creating an additional object for borderTiles? Can you perhaps just feed room.GetBoundaryTiles() straight into UnionWith? Also, if past its initial creation innerTiles contains all the tiles, it should be renamed, as I can see it easily creating issues down the road with someone using it and expecting it to only contain the innertiles (or at least recombining it with borderTiles, and wasting time doing a useless operation).